src/modules/fw_pos_control_l1/fw_pos_control_l1_params.c

```diff
@@ -854,3 +854,114 @@ PARAM_DEFINE_INT32(FW_GPSF_LT, 30);
  * @group Mission
  */
 PARAM_DEFINE_FLOAT(FW_GPSF_R, 15.0f);
+
+/**
+ * Wind-based airspeed scaling factor in Eco
+ *
+ * Multiplying this factor with the current absolute wind estimate gives the airspeed offset
+ * added to the setpoint (which is FW_AIRSPD_MIN) in Eco mode. This helps to make the
+ * system more robust against disturbances in high wind.
+ *
+ * setpoint = FW_AIRSPD_MIN + FW_LND_AIRSPD_SC * wind
+ *
+ * @unit norm
+ * @min 0
+ * @max 2
+ * @decimal 2
+ * @increment 0.01
+ * @group FW TECS
+ */
+PARAM_DEFINE_FLOAT(FW_WIND_ARSP_SC, 0.0f);
+
+/**
+ * FW Speed mode setting
+ *
+ * Setting
+ *
+ * @min 0
+ * @max 4
+ * @value 0 Normal
+ * @value 1 Eco cruise
+ * @value 2 Eco full
+ * @value 3 Dash cruise
+ * @value 4 Dash full
+ * @group FW TECS
+ */
+PARAM_DEFINE_INT32(FW_SPD_MODE_SET, 0);
+
+
+/**
+ * Max altitude undershoot for Eco/Dash
+ *
+ * Eco/Dash mode is disabled if the current altitude is more than this value below the setpoint.
+ *
+ * @min 1.0
+ * @max 50.0
+ * @decimal 1
+ * @increment 0.1
+ * @group FW TECS
+ */
+PARAM_DEFINE_FLOAT(FW_ALT_ERR_U, 10.f);
+
+/**
+ * Max altitude overshoot for Eco/Dash
+ *
+ * Eco/Dash mode is disabled if the current altitude is more than this value above the setpoint.
+ *
+ * @min 1.0
+ * @max 50.0
+ * @decimal 1
+ * @increment 0.1
+ * @group FW TECS
+ */
+PARAM_DEFINE_FLOAT(FW_ALT_ERR_O, 20.f);
+
+/**
+ * Min altitude for Eco/Dash
+ *
+ * Eco/Dash mode can be enabled if above this relative altitude to home.
+ *
+ * @min -200.0
```

Comment:
@sfuhrer Does this make sense?

Reply:
Well I guess it should be supported to fly Eco below the takeoff location no? E.g. taking off from a mountain.